storage-local/src/main/java/io/kestra/storage/local/LocalStorage.java

```diff
@@ -32,14 +32,19 @@ public LocalStorage(LocalConfig config) throws IOException {
     }
 
     private Path getPath(String tenantId, URI uri) {
+        Path basePath = tenantId == null ? config.getBasePath().toAbsolutePath()
+            : Paths.get(config.getBasePath().toAbsolutePath().toString(), tenantId);
+        if(uri == null) {
+            return basePath;
+        }
+
         parentTraversalGuard(uri);
-        return tenantId == null ? Paths.get(config.getBasePath().toAbsolutePath().toString(), uri.getPath())
-            : Paths.get(config.getBasePath().toAbsolutePath().toString(), tenantId, uri.getPath());
+        return Paths.get(basePath.toString(), uri.getPath());
     }
 
     @Override
     public InputStream get(String tenantId, URI uri) throws IOException {
-        return new BufferedInputStream(new FileInputStream(getPath(tenantId, URI.create(uri.getPath()))
+        return new BufferedInputStream(new FileInputStream(getPath(tenantId, uri)
```

Comment:
I also be disturbed by these `URI.create(uri.getPath)` but they should be kept as they remove the Kestra scheme if any.
So go back on this, and maybe add a comment to explain why we do this at least on the first time we do this so nobody else will wonder why ;)

In fact, I'm surprised that the LocalStorage should be updated? Didn't it creates subdir?

Reply:
uri should be nullable to reference root path instead of a hacky URI.create("/") or "" so I modified it so that it works.
For the scheme every call I removed were within a `getPath()` call so it already removes the schema as it takes only uri.getPath in it so it's a duplicate